src/Runner/FileLintingIterator.php

```diff
@@ -48,18 +48,14 @@ public function next()
     {
         parent::next();
 
-        if ($this->valid()) {
-            $this->currentResult = $this->handleItem($this->current());
-        }
+        $this->currentResult = $this->valid() ? $this->handleItem($this->current()) : null;
```

Comment:
what issue you experienced without setting this value to `null` if position is not valid?
also, looks like typehint of `$currentResult` property is wrong

Reply:
>what issue you experienced without setting this value to null if position is not valid?

when not valid getting the current value returns the previous valid value, which is unexpected

>also, looks like typehint of $currentResult property is wrong

I forgot to update to `|null` indeed